src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/BinderHelper.cs

```diff
@@ -195,7 +195,10 @@ internal static bool IsWindowsRuntimeObject(DynamicMetaObject obj)
 
         private static bool IsTransparentProxy(object obj)
         {
-            return obj != null; 
```

Comment:
This looks like just a logical bug in trying to keep the part of the check unaffected by absence of RemotingServices.IsTransparentProxy